modules/social_features/social_content_report/src/Controller/ReportController.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Drupal\social_content_report\Controller;
+
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\flag\FlaggingInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+
+/**
+ * Class ReportController.
+ *
+ * @package Drupal\social_content_report\Controller
+ */
+class ReportController extends ControllerBase {
+
+  /**
+   * Function for suggestions.
+   *
+   * @param \Drupal\flag\FlaggingInterface $flagging
+   *   The Flagging object to close.
+   *
+   * @return \Symfony\Component\HttpFoundation\RedirectResponse
+   *   A simple response.
+   *
+   * @throws \Drupal\Core\Entity\EntityStorageException
+   */
+  public function close(FlaggingInterface $flagging) {
+
+    if ($flagging->hasField('field_status')) {
+      // Disable the status field.
+      $flagging->set('field_status', 0);
+      $flagging->save();
+    }
+
+    return new RedirectResponse('view.report_overview.overview');
```

Comment:
It isn't needed because page which using an action of this callback adding a destination to links of this action.

Reply:
@OH17L I don't quite understand what you mean.

What I meant initially is if I send you to the `/report/:id/close?destination=node/20` URL then the user is still redirected to the Report Overview instead of to node with ID 20.

@bramtenhove Given time constraints I think we should create a follow-up here?